contracts/FeralfileArtworkV4_2.sol

```diff
@@ -0,0 +1,188 @@
+
+    error TokenIDNotFound();
+    error MintNotEnabled();
+
+    struct TokenInfo {
+        string thumbnail;
+        bytes parameters;
+    }
+
+    struct MintDataWithIndex {
+        uint256 seriesId;
+        uint256 tokenId;
+        address owner;
+        uint256 tokenIndex;
+    }
+
+    mapping(uint256 => TokenInfo) public _tokenInfos; // tokenID => tokenInfo
+    mapping(uint256 => uint256) public _tokenIndexes; // tokenID => tokenIndex
+
+    string public tokenPlaceholderURL;
+    string public tokenPlaceholderThubmnail;
+
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        bool burnable_,
+        bool bridgeable_,
+        address signer_,
+        address vault_,
+        address costReceiver_,
+        string memory contractURI_,
+        uint256[] memory seriesIds_,
+        uint256[] memory seriesMaxSupplies_
+    )
+        FeralfileExhibitionV4_1(
+            name_,
+            symbol_,
+            burnable_,
+            bridgeable_,
+            signer_,
+            vault_,
+            costReceiver_,
+            contractURI_,
+            seriesIds_,
+            seriesMaxSupplies_
+        )
+    {}
+    
+    /// @notice Mint new collection of Artwork
+    /// @dev the function iterates over the array of MintDataWithIndex to call the internal function mintArtworksWithIndex
+    /// @param data an array of MintDataWithIndex
+    function mintArtworksWithIndex(
+        MintDataWithIndex[] calldata data
+    ) external onlyAuthorized {
+        if (!mintable) {
+            revert MintNotEnabled();
+        }
+
+        for (uint256 i = 0; i < data.length; i++) {
+            _tokenIndexes[data[i].tokenId] = data[i].tokenIndex;
+            _mintArtwork(data[i].seriesId, data[i].tokenId, data[i].owner);
+        }
+    }
+
+    /// @notice Update the thumbnail & parameters of an edition to a new value
+    function updateTokenInformation(uint256 tokenId, string calldata thumbnail, bytes calldata parameters)
+        external
+        onlyAuthorized
+    {
+        if (!_exists(tokenId)) {
+            revert TokenIDNotFound();
+        }
+
+        _tokenInfos[tokenId] = TokenInfo(thumbnail, parameters);
+    }
+
+    /// @notice _buildArtworkData returns an object of artwork which would push to the actually artwork
+    /// @param parameters - the parameters for building artwork data
+    function _buildArtworkData(bytes memory parameters)
+        private
+        pure
+        returns (string memory)
+    {
+        return Base64.encode(parameters);
+    }
+
+    /// @notice _buildPreviewURL returns the preview url
+    /// @param parameters - the token paramters
+    function _buildPreviewURL(bytes memory parameters)
+        private
+        view
+        returns (string memory)
+    {
+        if (parameters.length == 0) {
+            return tokenPlaceholderURL;
+        }
+
+        return _buildIframe(_buildArtworkData(parameters), tokenBaseURI);
```

Comment:
`tokenBaseURI` is quite confusing since it actually indicates the token URI, not an artwork file URI we're thinking. 
To reduce the confusion, I think we should implement the `setTokenBaseURI` and throw error there. Instead, we could create a new variable to store `artworkFileURI` and allow to set by authororized accounts.